contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -545,41 +491,24 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
     /// @param tokenAddressOut Address of the token to get after conversion
     /// @param to Address of the tokenAddressOut receiver
     /// @return actualAmountIn Actual amount of tokenAddressIn transferred
-    /// @return amountConvertedMantissa Amount of tokenAddressIn supposed to get transferred
     /// @return actualAmountOut Actual amount of tokenAddressOut transferred
-    /// @return amountOutMantissa Amount of tokenAddressOut supposed to get transferred
     /// @custom:error AmountOutLowerThanMinRequired error is thrown when amount of output tokenAddressOut is less than amountOutMinMantissa
     function _convertExactTokensForTokens(
         uint256 amountInMantissa,
         uint256 amountOutMinMantissa,
         address tokenAddressIn,
         address tokenAddressOut,
         address to
-    )
-        internal
-        returns (
-            uint256 actualAmountIn,
-            uint256 amountConvertedMantissa,
-            uint256 actualAmountOut,
-            uint256 amountOutMantissa
-        )
-    {
-        (amountConvertedMantissa, amountOutMantissa) = getUpdatedAmountOut(
-            amountInMantissa,
-            tokenAddressIn,
-            tokenAddressOut
-        );
+    ) internal returns (uint256 actualAmountIn, uint256 actualAmountOut) {
+        actualAmountIn = _actualAmountIn(tokenAddressIn, amountInMantissa);
+
+        (, uint256 amountOutMantissa) = getUpdatedAmountOut(actualAmountIn, tokenAddressIn, tokenAddressOut);
```

Comment:
I think we have to take into account the first returned variable (`amountConvertedMantissa`), and compare it with `actualAmountIn`. If `actualAmountIn > amountConvertedMantissa`, we should refund the difference to the sender

Reply:
`getUpdatedAmountOut` will return always `amountConvertedMantissa` equal to `actualAmountIn`. So, my suggested check doesn't make sese